rtc/encryption.md

```diff
@@ -0,0 +1,70 @@
+
+Document collaboration may be performed in multiple sessions. For example, when a new version of {{site.productname}} is deployed it may be incompatible with existing sessions. Only one session will be active at a time but older sessions may still be used to bootstrap new sessions. As such, old keys cannot be immediately discarded when a new key is requested.
+
+In order to allow for key rotation a key hint is supplied so the encryption provider may tell the difference between these two cases and act accordingly. If the key hint is `null`, then the RTC protocol needs a new key and this is the opportunity for key rotation. If the key hint is not null the protocol is requesting a previously issued key so that it can read the session history.
+
+If the key hint is not included in an encryption provider response for a new key, an empty string will be recorded as the key hint.
+
+> **Warning**: The key hint is stored by the server in plain text. Do not add secret or sensitive information to the key hint.
+
+The key hint can be a key thumbprint, ID, or other non-sensitive identifier that will help select the key, such as a timestamp. It is only recorded when the protocol requests a new key. Returning a key hint from requests for previously issued keys will have no effect.
+
+## RTC encryption details
+
+> **Note**: This section contains the technical details of encryption used to securely transmit document contents. It is provided for information purposes only; understanding these details is not required to integrate RTC.
+
+The RTC protocol does not use the [provided encryption key]({{site.baseurl}}/rtc/configuration#rtc_encryption_provider) to encrypt content directly. Using industry standard cryptography algorithms a unique session key for content encryption is derived from the provided key in a way that makes brute force decryption very difficult.
+
+> **Note**: {{site.companyname}} is in the process of changing RTC encryption to [AES-KW](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/wrapKey) which does not require an initialization vector or protocol salt. The description below is still correct for the current beta release.
+
+The RTC protocol encryption technique is as follows:
+* 256 bits of random data are generated as the salt for each session. As described above, each document ID used for collaboration may have multiple sessions. The salt is generated with the [Crypto.getRandomValues()](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/getRandomValues) browser API and a `Uint8Array` of length 32 bytes.
+* The salt data and the provided encryption key are combined using the [PBKDF2](https://en.wikipedia.org/wiki/PBKDF2) key derivation function. Derivation is performed by the [SubtleCrypto.deriveKey()](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/deriveKey) browser API, using the following parameters:
```

Comment:
Regarding the term "key derivation function" and "derivation", we need to either:
- link to a reliable source
- provide a description
- avoid using
I know this is a technical section, but that also means it's easy to get lost.

Reply:
wikipedia isn't reliable?

Perhaps this entire section needs to be shifted to a dedicated "security professionals, here is how we can prove to you we know what we're talking about" page instead of tacking it onto the end of the general integrator-targeted encryption page.

Reply:
> wikipedia isn't reliable?

no. it's not.

Regarding your second point, maybe? This is something we could try address now or before the RTC 1.0 release.

Reply:
I could link to the IETF RFC, but that's _completely_ unintelligible. Wikipedia at least gives an average user a chance if they try to follow this.

I think moving it to a separate page can be a pre-1.0 task, rather than trying to figure it out now.

Reply:
Google for PBKDF2 also brings up https://cryptobook.nakov.com/mac-and-key-derivation/pbkdf2 but that's someone's personal site by the looks of it. A professional someone, but I don't know if it will meet your standard for reliability.

Reply:
I went with the IETF. Google will bring up wikipedia, we don't need to link to it.